### PR TITLE
Use xkbcommon for keyboard event handling

### LINKED ISSUE #41: Cancel with actual Escape key
Support was added in #20 to cancel colour picking with the Escape key.
This is mapped to the physical Escape key on most keyboards, which is not always the actual Escape key
(eg. I have swapped Escape and Caps Lock on my keyboard).

### PR DESCRIPTION
**Pull request by TheOnlyMrCat**
Fixes #41.

This is my first time writing C++ in a long time, so I might have put some things in the wrong place.

Errors in initialising xkb are logged, but not treated as fatal. If we can't initialise xkb, we just don't handle keypresses at all.

### CODE REVIEW

File: src/hyprpicker.cpp [hunk 1]
```diff
@@ -8,6 +8,12 @@ void sigHandler(int sig) {
 }
 
 void CHyprpicker::init() {
+    m_pXKBContext = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
+    if (!m_pXKBContext) {
```
Comment (vaxerski) [inline src/hyprpicker.cpp:12]: style: no {} around short ifs

File: src/events/Events.cpp [hunk 2]
```diff
@@ -246,18 +246,51 @@ void Events::handlePointerButton(void* data, struct wl_pointer* wl_pointer, uint
...
+    g_pHyprpicker->m_pXKBState = xkb_state_new(g_pHyprpicker->m_pXKBKeymap);
+    if (!g_pHyprpicker->m_pXKBState) {
+        Debug::log(ERR, "Failed to create xkb state");
+        return;
+    }
+}
 
 void Events::handleKeyboardKey(void* data, struct wl_keyboard* keyboard, uint32_t serial, uint32_t time, uint32_t key, uint32_t state) {
-    if (key == 1) // escape
+    if (!g_pHyprpicker->m_pXKBState || state != WL_KEYBOARD_KEY_STATE_PRESSED)
```
Comment (vaxerski) [inline src/events/Events.cpp:277]: I would say if the state failed, we shouldn't ignore keys altogether, but fall back to raw keycodes. Probably log it too.

File: src/hyprpicker.cpp [hunk 3]
```diff
@@ -8,6 +8,11 @@ void sigHandler(int sig) {
 }
 
 void CHyprpicker::init() {
+    m_pXKBContext = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
+    if (!m_pXKBContext)
+        Debug::log(ERR, "Failed to create xkb context");
+    m_pXKBState = nullptr;
```
Comment (vaxerski) [inline src/hyprpicker.cpp:14]: instead of this, I'd initialize all 3 xkb_ ptrs to nullptr in da header

File: src/events/Events.cpp (hunk 2 above)
Comment (TheOnlyMrCat) [reply src/events/Events.cpp]: I'm outputting logs if any of the initialisation steps fail; is that sufficient?

Comment (vaxerski) [reply src/events/Events.cpp]: ye

**Review by vaxerski — APPROVED** [review]
lgtm, I am a coconut

**Comment by vaxerski** [comment]
thanks!

**Pull request merged**